theano/tensor/tests/test_basic.py

```diff
@@ -73,8 +73,34 @@
 else:
     mode_opt = get_default_mode()
 
-### seed random number generator so that unittests are deterministic ###
+
+# Use a seeded random number generator so that unittests are deterministic
 utt.seed_rng()
+test_rng = numpy.random.RandomState(seed=utt.fetch_seed())
+
+
+# A helpful class to check random values close to the boundaries
+# when designing new tests
+class MockRandomState:
```

Comment:
I'm pushing in this PR to allow merging this rapidly for rc4. I'm moving this to theano/tests/unittest_tools.py

If CI pass, I'll merge tomorrow morning.